components/admin_console/reset_email_modal/reset_email_modal.tsx

```diff
@@ -48,38 +61,38 @@ export default class ResetEmailModal extends React.PureComponent {
                 this.setState({error: errMsg});
                 return;
             }
+            email = email.trim().toLowerCase();
         }
 
-        const user = Object.assign({}, this.props.user);
-        const email = this.emailRef.current.value.trim().toLowerCase();
-        user.email = email;
-
-        this.setState({error: null});
-
-        adminResetEmail(
-            user,
-            () => {
-                this.props.onModalSubmit(this.props.user);
-            },
-            (err) => {
-                const serverError = err.message ? err.message : err;
-                this.setState({error: serverError});
-            },
-        );
+        const user = {
+            ...this.props.user,
+            email,
+        };
+
+        const result = await this.props.actions.patchUser(user);
+        if ('error' in result) {
+            this.setState({error: result.error.message});
+            return;
+        }
+        this.props.onModalSubmit(this.props.user);
     }
 
-    doCancel = () => {
-        this.setState({error: null});
+    private doCancel = (): void => {
+        this.setState({
+            error: null,
+        });
         this.props.onModalDismissed();
     }
 
-    render() {
-        if (!this.props.user) {
+    public render(): JSX.Element {
+        const user = this.props.user;
+        if (user == null) {
```

Comment:
Is there a reason we can do `!user` here?

Reply:
Yes, I already fix it.